products/tpu/terraform.yaml

```diff
@@ -36,10 +36,14 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       network: !ruby/object:Overrides::Terraform::PropertyOverride
         default_from_api: true
         diff_suppress_func: 'compareSelfLinkOrResourceName'
+      cidrBlock: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
       schedulingConfig: !ruby/object:Overrides::Terraform::PropertyOverride
         diff_suppress_func: 'compareTpuNodeSchedulingConfig'
       schedulingConfig.preemptible: !ruby/object:Overrides::Terraform::PropertyOverride
         diff_suppress_func: 'compareTpuNodeSchedulingConfig'
+      network: !ruby/object:Overrides::Terraform::PropertyOverride
```

Comment:
This seems to be overwriting the `network` block from line 36, and also, I don't think we want to always suppress this diff. What if the intent is to change the network, I don't think it'd recognize the change.

Reply:
Reworked it to check for a change in network.

Reply:
I'm not sure this will make a huge difference, because if the user really wants to change the network, this will prevent them from doing so. I wonder if we could maybe do a diffsuppress that would parse the network and see if the only difference is the project and then check that the project number is the _correct_ project number? Does that make sense? It's a little hacky, but would be the workaround until the upstream issue was solved.
Also, I think we want to use the `diff_suppress_func` from line 38 above rather than creating a new `network` block here.  Then, in the new diff suppress func that you're implementing, make sure you include the same logic that is in `compareSelfLinkOrResourceName` so we don't lose that as well. Does that make sense?
Let me know if you have questions.

Reply:
Have a look my last commit. How do I get the `*Config` in to diffsupress function? I'm using other functions that expect the config block as arguments.

Reply:
Hi @upodroid! What a finicky issue! I spent my morning looking into how we can do this. Although diff suppress would be ideal, it looks like we can't use it here. If we want access to `config` we'll have to use a `CustomizeDiff`.
Basically what I did in the CustomizeDiff was get the `old` (value using project number) and `new` (value using project name) values, used regex to find the submatch for project.  Then called out to the projects API using the project name and if the returned `project.Number` was the same as the project number, then i did a `diff.SetNew(old)`. Does that make sense? So basically if they're the same project, we're just going to set it to use the project number so there is no diff.
Let me know if you have questions! Thanks again!!

Reply:
I amended it to use CustomizeDiff now but i'm still getting diffs.

Reply:
nvm. It is ready to go.

```
 REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google-beta   master  66⬇  5✎  USAGE  $    make testacc TEST=./google-beta TESTARGS='-run=TestAccTPUNode_tpuNodeFullExample'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccTPUNode_tpuNodeFullExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccTPUNode_tpuNodeFullExample
=== PAUSE TestAccTPUNode_tpuNodeFullExample
=== CONT  TestAccTPUNode_tpuNodeFullExample
--- PASS: TestAccTPUNode_tpuNodeFullExample (364.17s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta 366.543s
```